Cromwell does not exit in single workflow mode, using JES, on successful workflow completion

- JES backend

```
...snip....
[2016-11-03 19:36:22,19] [info] SingleWorkflowRunnerActor writing metadata to /home/lichtens/test_eval/crsp_validation_input_files/crsp_validation_from_cromwell.json.metadata.json
[2016-11-03 19:36:22,30] [info] SingleWorkflowRunnerActor workflow finished with status 'Succeeded'.
[2016-11-03 19:36:22,30] [info] WorkflowManagerActor: Received shutdown signal.
[2016-11-03 19:36:22,30] [info] Waiting for 1 workflows to abort...
....15 minutes gone by....
```

I upgraded to release 0.23 and did a workaround for #1754 ...

I had the same issue as described here, however, I did see the exception below, which I do not think was displayed in previous versions of cromwell.  Is this at all helpful?  @kshakir ?  I'd also like to point out that this file does not exist, but I have not done anything to that directory.  There are log files from other runs (mostly local backend, though).  The permissions are set appropriately for ``/home/lichtens/eval-gatk-protected/cromwell-workflow-logs/``.

```
[ERROR] [12/07/2016 22:51:59.735] [cromwell-system-akka.dispatchers.engine-dispatcher-53] [akka://cromwell-system/user/SingleWorkflowRunnerActor/WorkflowManagerActor/WorkflowActor-8968c364-
3623-4242-b39e-228f43f5d4c3] /home/lichtens/eval-gatk-protected/cromwell-workflow-logs/workflow.8968c364-3623-4242-b39e-228f43f5d4c3.log
java.nio.file.NoSuchFileException: /home/lichtens/eval-gatk-protected/cromwell-workflow-logs/workflow.8968c364-3623-4242-b39e-228f43f5d4c3.log
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:244)
        at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
        at java.nio.file.Files.delete(Files.java:1126)
        at better.files.File.delete(File.scala:602)
        at cromwell.core.logging.WorkflowLogger$$anonfun$deleteLogFile$1.apply(WorkflowLogger.scala:112)
        at cromwell.core.logging.WorkflowLogger$$anonfun$deleteLogFile$1.apply(WorkflowLogger.scala:112)
        at scala.Option.foreach(Option.scala:257)
        at cromwell.core.logging.WorkflowLogger.deleteLogFile(WorkflowLogger.scala:112)
        at cromwell.engine.workflow.WorkflowActor$$anonfun$9$$anonfun$applyOrElse$1.apply(WorkflowActor.scala:307)
        at cromwell.engine.workflow.WorkflowActor$$anonfun$9$$anonfun$applyOrElse$1.apply(WorkflowActor.scala:303)
        at scala.Option.foreach(Option.scala:257)
        at cromwell.engine.workflow.WorkflowActor$$anonfun$9.applyOrElse(WorkflowActor.scala:303)
        at cromwell.engine.workflow.WorkflowActor$$anonfun$9.applyOrElse(WorkflowActor.scala:288)
        at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
        at akka.actor.FSM$$anonfun$handleTransition$1.apply(FSM.scala:606)
        at akka.actor.FSM$$anonfun$handleTransition$1.apply(FSM.scala:606)
        at scala.collection.immutable.List.foreach(List.scala:381)
        at akka.actor.FSM$class.handleTransition(FSM.scala:606)
        at akka.actor.FSM$class.makeTransition(FSM.scala:688)
        at cromwell.engine.workflow.WorkflowActor.makeTransition(WorkflowActor.scala:154)
        at akka.actor.FSM$class.applyState(FSM.scala:673)
        at cromwell.engine.workflow.WorkflowActor.applyState(WorkflowActor.scala:154)
        at akka.actor.FSM$class.processEvent(FSM.scala:668)
        at cromwell.engine.workflow.WorkflowActor.akka$actor$LoggingFSM$$super$processEvent(WorkflowActor.scala:154)
        at akka.actor.LoggingFSM$class.processEvent(FSM.scala:799)
        at cromwell.engine.workflow.WorkflowActor.processEvent(WorkflowActor.scala:154)
        at akka.actor.FSM$class.akka$actor$FSM$$processMsg(FSM.scala:657)
        at akka.actor.FSM$$anonfun$receive$1.applyOrElse(FSM.scala:651)
        at akka.actor.Actor$class.aroundReceive(Actor.scala:484)
        at cromwell.engine.workflow.WorkflowActor.aroundReceive(WorkflowActor.scala:154)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
        at akka.actor.ActorCell.invoke(ActorCell.scala:495)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
        at akka.dispatch.Mailbox.run(Mailbox.scala:224)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)


```

--- a/backend/src/main/scala/cromwell/backend/callcaching/CacheHitDuplicating.scala
+++ b/backend/src/main/scala/cromwell/backend/callcaching/CacheHitDuplicating.scala
@@ -7,6 +7,7 @@ import cromwell.backend.BackendCacheHitCopyingActor
 import cromwell.backend.BackendJobExecutionActor.{BackendJobExecutionResponse, JobSucceededResponse}
 import cromwell.backend.io.JobPaths
 import cromwell.core.path.PathCopier
+import cromwell.core.path.PathImplicits._
 import cromwell.core.simpleton.{WdlValueBuilder, WdlValueSimpleton}
 import wdl4s.values.WdlFile
 
@@ -66,7 +67,7 @@ trait CacheHitDuplicating {
         val sourcePath = getPath(wdlFile.value).get
         val destinationPath = PathCopier.getDestinationFilePath(sourceCallRootPath, sourcePath, destinationCallRootPath)
         duplicate(sourcePath, destinationPath)
-        WdlValueSimpleton(key, WdlFile(destinationPath.toUri.toString))
+        WdlValueSimpleton(key, WdlFile(destinationPath.toRealString))
       case wdlValueSimpleton => wdlValueSimpleton
     }
   }

--- a/core/src/main/scala/cromwell/core/logging/WorkflowLogger.scala
+++ b/core/src/main/scala/cromwell/core/logging/WorkflowLogger.scala
@@ -14,6 +14,8 @@ import net.ceedubs.ficus.Ficus._
 import org.slf4j.helpers.NOPLogger
 import org.slf4j.{Logger, LoggerFactory}
 
+import scala.util.Try
+
 trait WorkflowLogging extends ActorLogging { this: Actor =>
   def workflowIdForLogging: WorkflowId
 
@@ -109,7 +111,7 @@ class WorkflowLogger(loggerName: String,
 
   override def getName = loggerName
 
-  def deleteLogFile() = workflowLogPath foreach { File(_).delete(swallowIOExceptions = false) }
+  def deleteLogFile() = Try { workflowLogPath foreach { File(_).delete(swallowIOExceptions = false) } }
 
   import WorkflowLogger._
 

--- a/core/src/main/scala/cromwell/core/path/PathFactory.scala
+++ b/core/src/main/scala/cromwell/core/path/PathFactory.scala
@@ -46,7 +46,7 @@ object PathFactory {
                 postMapping: Path => Path = identity[Path]): Path = {
     pathBuilders.toStream map { _.build(preMapping(string)) } collectFirst { case Success(p) => postMapping(p) } getOrElse {
       val pathBuilderNames: String = pathBuilders map { _.name } mkString ", "
-      throw new PathParsingException(s"Could not find suitable filesystem among $pathBuilderNames to parse $string.")
+      throw PathParsingException(s"Could not find suitable filesystem among $pathBuilderNames to parse $string.")
     }
   }
 

--- a/core/src/main/scala/cromwell/core/path/PathImplicits.scala
+++ b/core/src/main/scala/cromwell/core/path/PathImplicits.scala
@@ -11,5 +11,7 @@ object PathImplicits {
     def untailed = UntailedWriter(path)
 
     def tailed(tailedSize: Int) = TailedWriter(path, tailedSize)
+
+    def toRealString: String = java.net.URLDecoder.decode(path.toUri.toString, "UTF-8")
   }
 }

--- a/core/src/main/scala/cromwell/core/path/proxy/RetryableFileSystemProviderProxy.scala
+++ b/core/src/main/scala/cromwell/core/path/proxy/RetryableFileSystemProviderProxy.scala
@@ -37,7 +37,9 @@ class RetryableFileSystemProviderProxy[T <: FileSystemProvider](delegate: T, ret
     new FileSystemProxy(delegate.newFileSystem(uri, env), this)
   }
   override def getScheme: String = delegate.getScheme
-  override def getFileSystem(uri: URI): FileSystem = delegate.getFileSystem(uri)
+  override def getFileSystem(uri: URI): FileSystem = {
+    new FileSystemProxy(delegate.getFileSystem(uri), this)
+  }
   override def getFileStore(path: Path): FileStore = delegate.getFileStore(path)
 
   /* retried operations */

--- a/engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala
+++ b/engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala
@@ -23,6 +23,8 @@ import cromwell.subworkflowstore.SubWorkflowStoreActor.WorkflowComplete
 import cromwell.webservice.EngineStatsActor
 import wdl4s.{LocallyQualifiedName => _}
 
+import scala.util.Failure
+
 object WorkflowActor {
 
   /**
@@ -315,7 +317,10 @@ class WorkflowActor(val workflowId: WorkflowId,
         workflowOptions.get(FinalWorkflowLogDir).toOption match {
             case Some(destinationDir) =>
               workflowLogCopyRouter ! CopyWorkflowLogsActor.Copy(workflowId, PathFactory.buildPath(destinationDir, pathBuilders))
-            case None if WorkflowLogger.isTemporary => workflowLogger.deleteLogFile()
+            case None if WorkflowLogger.isTemporary => workflowLogger.deleteLogFile() match {
+              case Failure(f) => log.error(f, "Failed to delete workflow log")
+              case _ =>
+            }
             case _ =>
           }
         }

--- a/engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCache.scala
+++ b/engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCache.scala
@@ -8,6 +8,7 @@ import cromwell.core.ExecutionIndex.IndexEnhancedIndex
 import cromwell.core.WorkflowId
 import cromwell.core.callcaching.HashResult
 import cromwell.core.simpleton.WdlValueSimpleton
+import cromwell.core.path.PathImplicits._
 import cromwell.database.sql._
 import cromwell.database.sql.joins.CallCachingJoin
 import cromwell.database.sql.tables.{CallCachingDetritusEntry, CallCachingEntry, CallCachingHashEntry, CallCachingSimpletonEntry}
@@ -53,7 +54,7 @@ class CallCache(database: CallCachingSqlDatabase) {
 
     val jobDetritusToInsert: Iterable[CallCachingDetritusEntry] = {
       jobDetritus map {
-        case (fileName, filePath) => CallCachingDetritusEntry(fileName, filePath.toUri.toString)
+        case (fileName, filePath) => CallCachingDetritusEntry(fileName, filePath.toRealString)
       }
     }
 

--- a/filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsPathBuilder.scala
+++ b/filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsPathBuilder.scala
@@ -11,9 +11,11 @@ import com.google.cloud.RetryParams
 import com.google.cloud.storage.StorageOptions
 import com.google.cloud.storage.contrib.nio.{CloudStorageConfiguration, CloudStorageFileSystem, CloudStoragePath}
 import com.google.common.base.Preconditions._
+import com.google.common.net.UrlEscapers
 import cromwell.core.WorkflowOptions
 import cromwell.core.path.proxy.{PathProxy, RetryableFileSystemProviderProxy}
 import cromwell.core.path.{CustomRetryParams, PathBuilder}
+import cromwell.filesystems.gcs.GcsPathBuilder._
 import cromwell.filesystems.gcs.auth.GoogleAuthMode
 
 import scala.util.{Failure, Try}
@@ -35,12 +37,14 @@ object GcsPathBuilder {
   }
 
   def isValidGcsUrl(str: String): Boolean = {
-    Try(checkValid(URI.create(str))).isSuccess
+    Try(checkValid(getUri(str))).isSuccess
   }
 
   def isGcsPath(path: Path): Boolean = {
     path.getFileSystem.provider().getScheme == CloudStorageFileSystem.URI_SCHEME
   }
+
+  def getUri(string: String) = URI.create(UrlEscapers.urlFragmentEscaper().escape(string))
 }
 
 class GcsPathBuilder(authMode: GoogleAuthMode,
@@ -78,7 +82,7 @@ class GcsPathBuilder(authMode: GoogleAuthMode,
 
   def build(string: String): Try[Path] = {
     Try {
-      val uri = URI.create(string)
+      val uri = getUri(string)
       GcsPathBuilder.checkValid(uri)
       provider.getPath(uri)
     }

--- a/supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala
+++ b/supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala
@@ -25,6 +25,7 @@ import cromwell.core._
 import cromwell.core.logging.JobLogging
 import cromwell.core.path.proxy.PathProxy
 import cromwell.core.retry.{Retry, SimpleExponentialBackoff}
+import cromwell.core.path.PathImplicits._
 import cromwell.services.keyvalue.KeyValueServiceActor._
 import cromwell.services.metadata._
 import wdl4s._
@@ -101,9 +102,9 @@ class JesAsyncBackendJobExecutionActor(override val jobDescriptor: BackendJobDes
 
   override lazy val retryable = jobDescriptor.key.attempt <= runtimeAttributes.preemptible
   private lazy val cmdInput =
-    JesFileInput(ExecParamName, jesCallPaths.script.toUri.toString, Paths.get(jesCallPaths.scriptFilename), workingDisk)
+    JesFileInput(ExecParamName, jesCallPaths.script.toRealString, Paths.get(jesCallPaths.scriptFilename), workingDisk)
   private lazy val jesCommandLine = s"/bin/bash ${cmdInput.containerPath}"
-  private lazy val rcJesOutput = JesFileOutput(returnCodeFilename, returnCodeGcsPath.toUri.toString, Paths.get(returnCodeFilename), workingDisk)
+  private lazy val rcJesOutput = JesFileOutput(returnCodeFilename, returnCodeGcsPath.toRealString, Paths.get(returnCodeFilename), workingDisk)
 
   private lazy val standardParameters = Seq(rcJesOutput)
   private lazy val returnCodeContents = Try(File(returnCodeGcsPath).contentAsString)
@@ -129,14 +130,14 @@ class JesAsyncBackendJobExecutionActor(override val jobDescriptor: BackendJobDes
 
   private def gcsAuthParameter: Option[JesInput] = {
     if (jesAttributes.auths.gcs.requiresAuthFile || dockerConfiguration.isDefined)
-      Option(JesLiteralInput(ExtraConfigParamName, jesCallPaths.gcsAuthFilePath.toUri.toString))
+      Option(JesLiteralInput(ExtraConfigParamName, jesCallPaths.gcsAuthFilePath.toRealString))
     else None
   }
 
   private lazy val callContext = CallContext(
     callRootPath,
-    jesStdoutFile.toUri.toString,
-    jesStderrFile.toUri.toString
+    jesStdoutFile.toRealString,
+    jesStderrFile.toRealString
   )
 
   private[jes] lazy val callEngineFunctions = new JesExpressionFunctions(List(jesCallPaths.gcsPathBuilder), callContext)
@@ -233,7 +234,7 @@ class JesAsyncBackendJobExecutionActor(override val jobDescriptor: BackendJobDes
   }
 
   private def generateJesSingleFileOutputs(wdlFile: WdlSingleFile): JesFileOutput = {
-    val destination = callRootPath.resolve(wdlFile.value.stripPrefix("/")).toUri.toString
+    val destination = callRootPath.resolve(wdlFile.value.stripPrefix("/")).toRealString
     val (relpath, disk) = relativePathAndAttachedDisk(wdlFile.value, runtimeAttributes.disks)
     JesFileOutput(makeSafeJesReferenceName(wdlFile.value), destination, relpath, disk)
   }
@@ -242,8 +243,8 @@ class JesAsyncBackendJobExecutionActor(override val jobDescriptor: BackendJobDes
     val globName = callEngineFunctions.globName(wdlFile.value)
     val globDirectory = globName + "/"
     val globListFile = globName + ".list"
-    val gcsGlobDirectoryDestinationPath = callRootPath.resolve(globDirectory).toUri.toString
-    val gcsGlobListFileDestinationPath = callRootPath.resolve(globListFile).toUri.toString
+    val gcsGlobDirectoryDestinationPath = callRootPath.resolve(globDirectory).toRealString
+    val gcsGlobListFileDestinationPath = callRootPath.resolve(globListFile).toRealString
 
     val (_, globDirectoryDisk) = relativePathAndAttachedDisk(wdlFile.value, runtimeAttributes.disks)
 
@@ -318,7 +319,7 @@ class JesAsyncBackendJobExecutionActor(override val jobDescriptor: BackendJobDes
       runIdForResumption,
       jobDescriptor = jobDescriptor,
       runtimeAttributes = runtimeAttributes,
-      callRootPath = callRootPath.toUri.toString,
+      callRootPath = callRootPath.toRealString,
       commandLine = jesCommandLine,
       logFileName = jesLogFilename,
       jesParameters,

--- a/supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesExpressionFunctions.scala
+++ b/supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesExpressionFunctions.scala
@@ -6,6 +6,7 @@ import cromwell.backend.wdl.{ReadLikeFunctions, WriteFunctions}
 import cromwell.core.CallContext
 import cromwell.core.path.PathBuilder
 import cromwell.filesystems.gcs.GcsPathBuilder
+import cromwell.core.path.PathImplicits._
 import wdl4s.expression.{PureStandardLibraryFunctionsLike, WdlStandardLibraryFunctions}
 import wdl4s.values._
 
@@ -24,11 +25,11 @@ class JesExpressionFunctions(override val pathBuilders: List[PathBuilder], conte
   override def glob(path: String, pattern: String): Seq[String] = {
     val name = globName(pattern)
     val listFile = context.root.resolve(s"$name.list").toRealPath()
-    Files.readAllLines(listFile).asScala map { fileName => context.root.resolve(s"$name/$fileName").toUri.toString }
+    Files.readAllLines(listFile).asScala map { fileName => context.root.resolve(s"$name/$fileName").toRealString }
   }
 
   override def preMapping(str: String): String = if (!GcsPathBuilder.isValidGcsUrl(str)) {
-    context.root.resolve(str.stripPrefix("/")).toUri.toString
+    context.root.resolve(str.stripPrefix("/")).toRealString
   } else str
 
   override def stdout(params: Seq[Try[WdlValue]]) = Success(WdlFile(context.stdout))

--- a/supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesJobCachingActorHelper.scala
+++ b/supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesJobCachingActorHelper.scala
@@ -8,6 +8,7 @@ import cromwell.backend.BackendWorkflowDescriptor
 import cromwell.backend.callcaching.JobCachingActorHelper
 import cromwell.backend.impl.jes.io.{JesAttachedDisk, JesWorkingDisk}
 import cromwell.core.logging.JobLogging
+import cromwell.core.path.PathImplicits._
 
 import scala.util.Try
 
@@ -61,7 +62,7 @@ trait JesJobCachingActorHelper extends JobCachingActorHelper {
   lazy val jesAttributes = jesConfiguration.jesAttributes
   lazy val monitoringScript: Option[JesInput] = {
     jesCallPaths.monitoringPath map { path =>
-      JesFileInput(s"$MonitoringParamName-in", path.toUri.toString,
+      JesFileInput(s"$MonitoringParamName-in", path.toRealString,
         JesWorkingDisk.MountPoint.resolve(JesMonitoringScript), workingDisk)
     }
   }

--- a/supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesWorkflowPaths.scala
+++ b/supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesWorkflowPaths.scala
@@ -9,6 +9,7 @@ import cromwell.backend.io.WorkflowPaths
 import cromwell.backend.{BackendJobDescriptorKey, BackendWorkflowDescriptor}
 import cromwell.core.WorkflowOptions
 import cromwell.core.path.PathBuilder
+import cromwell.core.path.PathImplicits._
 import cromwell.filesystems.gcs.{GcsPathBuilderFactory, RetryableGcsPathBuilder}
 
 import scala.language.postfixOps
@@ -42,7 +43,7 @@ class JesWorkflowPaths(val workflowDescriptor: BackendWorkflowDescriptor,
     // The default auth file bucket is always at the root of the root workflow
     val defaultBucket = executionRoot.resolve(workflowDescriptor.rootWorkflow.unqualifiedName).resolve(workflowDescriptor.rootWorkflowId.toString)
     
-    val bucket = workflowDescriptor.workflowOptions.get(JesWorkflowPaths.AuthFilePathOptionKey) getOrElse defaultBucket.toUri.toString
+    val bucket = workflowDescriptor.workflowOptions.get(JesWorkflowPaths.AuthFilePathOptionKey) getOrElse defaultBucket.toRealString
     val authBucket = GcsPathBuilderFactory(genomicsCredentials).withOptions(workflowOptions).build(bucket) recover {
       case ex => throw new Exception(s"Invalid gcs auth_bucket path $bucket", ex)
     } get

--- a/supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActorSpec.scala
+++ b/supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActorSpec.scala
@@ -15,6 +15,7 @@ import cromwell.backend.impl.jes.RunStatus.Failed
 import cromwell.backend.impl.jes.io.{DiskType, JesWorkingDisk}
 import cromwell.backend.impl.jes.statuspolling.JesApiQueryManager.DoPoll
 import cromwell.core.logging.JobLogger
+import cromwell.core.path.PathImplicits._
 import cromwell.core.{WorkflowId, WorkflowOptions, _}
 import cromwell.filesystems.gcs.GcsPathBuilderFactory
 import cromwell.filesystems.gcs.auth.GoogleAuthMode.NoAuthMode
@@ -593,13 +594,13 @@ class JesAsyncBackendJobExecutionActorSpec extends TestKitSuite("JesAsyncBackend
     val jesBackend = testActorRef.underlyingActor
 
     jesBackend.jesCallPaths.stdout should be(a[CloudStoragePath])
-    jesBackend.jesCallPaths.stdout.toUri.toString shouldBe
+    jesBackend.jesCallPaths.stdout.toRealString shouldBe
       "gs://path/to/gcs_root/wf_hello/e6236763-c518-41d0-9688-432549a8bf7c/call-hello/hello-stdout.log"
     jesBackend.jesCallPaths.stderr should be(a[CloudStoragePath])
-    jesBackend.jesCallPaths.stderr.toUri.toString shouldBe
+    jesBackend.jesCallPaths.stderr.toRealString shouldBe
       "gs://path/to/gcs_root/wf_hello/e6236763-c518-41d0-9688-432549a8bf7c/call-hello/hello-stderr.log"
     jesBackend.jesCallPaths.jesLogPath should be(a[CloudStoragePath])
-    jesBackend.jesCallPaths.jesLogPath.toUri.toString shouldBe
+    jesBackend.jesCallPaths.jesLogPath.toRealString shouldBe
       "gs://path/to/gcs_root/wf_hello/e6236763-c518-41d0-9688-432549a8bf7c/call-hello/hello.log"
   }
 
@@ -624,13 +625,13 @@ class JesAsyncBackendJobExecutionActorSpec extends TestKitSuite("JesAsyncBackend
     val jesBackend = testActorRef.underlyingActor
 
     jesBackend.jesCallPaths.stdout should be(a[CloudStoragePath])
-    jesBackend.jesCallPaths.stdout.toUri.toString shouldBe
+    jesBackend.jesCallPaths.stdout.toRealString shouldBe
       "gs://path/to/gcs_root/w/e6236763-c518-41d0-9688-432549a8bf7d/call-B/shard-2/B-2-stdout.log"
     jesBackend.jesCallPaths.stderr should be(a[CloudStoragePath])
-    jesBackend.jesCallPaths.stderr.toUri.toString shouldBe
+    jesBackend.jesCallPaths.stderr.toRealString shouldBe
       "gs://path/to/gcs_root/w/e6236763-c518-41d0-9688-432549a8bf7d/call-B/shard-2/B-2-stderr.log"
     jesBackend.jesCallPaths.jesLogPath should be(a[CloudStoragePath])
-    jesBackend.jesCallPaths.jesLogPath.toUri.toString shouldBe
+    jesBackend.jesCallPaths.jesLogPath.toRealString shouldBe
       "gs://path/to/gcs_root/w/e6236763-c518-41d0-9688-432549a8bf7d/call-B/shard-2/B-2.log"
   }
 

--- a/supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/JesCallPathsSpec.scala
+++ b/supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/JesCallPathsSpec.scala
@@ -2,6 +2,7 @@ package cromwell.backend.impl.jes
 
 import cromwell.backend.BackendSpec
 import cromwell.core.TestKitSuite
+import cromwell.core.path.PathImplicits._
 import cromwell.util.SampleWdl
 import org.scalatest.{FlatSpecLike, Matchers}
 import org.specs2.mock.Mockito
@@ -32,13 +33,13 @@ class JesCallPathsSpec extends TestKitSuite with FlatSpecLike with Matchers with
     val jesConfiguration = new JesConfiguration(JesBackendConfigurationDescriptor)
 
     val callPaths = JesJobPaths(jobDescriptorKey, workflowDescriptor, jesConfiguration)
-    callPaths.returnCode.toUri.toString should
+    callPaths.returnCode.toRealString should
       be(s"gs://my-cromwell-workflows-bucket/wf_hello/${workflowDescriptor.id}/call-hello/hello-rc.txt")
-    callPaths.stdout.toUri.toString should
+    callPaths.stdout.toRealString should
       be(s"gs://my-cromwell-workflows-bucket/wf_hello/${workflowDescriptor.id}/call-hello/hello-stdout.log")
-    callPaths.stderr.toUri.toString should
+    callPaths.stderr.toRealString should
       be(s"gs://my-cromwell-workflows-bucket/wf_hello/${workflowDescriptor.id}/call-hello/hello-stderr.log")
-    callPaths.jesLogPath.toUri.toString should
+    callPaths.jesLogPath.toRealString should
       be(s"gs://my-cromwell-workflows-bucket/wf_hello/${workflowDescriptor.id}/call-hello/hello.log")
   }
 
@@ -48,7 +49,7 @@ class JesCallPathsSpec extends TestKitSuite with FlatSpecLike with Matchers with
     val jesConfiguration = new JesConfiguration(JesBackendConfigurationDescriptor)
 
     val callPaths = JesJobPaths(jobDescriptorKey, workflowDescriptor, jesConfiguration)
-    callPaths.callContext.root.toUri.toString should
+    callPaths.callContext.root.toRealString should
       be(s"gs://my-cromwell-workflows-bucket/wf_hello/${workflowDescriptor.id}/call-hello")
     callPaths.callContext.stdout should be("hello-stdout.log")
     callPaths.callContext.stderr should be("hello-stderr.log")

--- a/supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/JesWorkflowPathsSpec.scala
+++ b/supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/JesWorkflowPathsSpec.scala
@@ -2,6 +2,7 @@ package cromwell.backend.impl.jes
 
 import cromwell.backend.BackendSpec
 import cromwell.core.TestKitSuite
+import cromwell.core.path.PathImplicits._
 import cromwell.util.SampleWdl
 import org.scalatest.{FlatSpecLike, Matchers}
 import org.specs2.mock.Mockito
@@ -17,10 +18,10 @@ class JesWorkflowPathsSpec extends TestKitSuite with FlatSpecLike with Matchers 
     val jesConfiguration = new JesConfiguration(JesBackendConfigurationDescriptor)
 
     val workflowPaths = JesWorkflowPaths(workflowDescriptor, jesConfiguration)(system)
-    workflowPaths.executionRoot.toUri.toString should be("gs://my-cromwell-workflows-bucket/")
-    workflowPaths.workflowRoot.toUri.toString should
+    workflowPaths.executionRoot.toRealString should be("gs://my-cromwell-workflows-bucket/")
+    workflowPaths.workflowRoot.toRealString should
       be(s"gs://my-cromwell-workflows-bucket/wf_hello/${workflowDescriptor.id}/")
-    workflowPaths.gcsAuthFilePath.toUri.toString should
+    workflowPaths.gcsAuthFilePath.toRealString should
       be(s"gs://my-cromwell-workflows-bucket/wf_hello/${workflowDescriptor.id}/${workflowDescriptor.id}_auth.json")
   }
 }

